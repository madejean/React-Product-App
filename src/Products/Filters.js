import React from 'react';

class Filters extends React.Component {
  constructor(props){
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(e) {
    const value = e.target[e.target.type === 'checkbox' ? 'checked' : 'value']
    const name = e.target.name;
    this.props.onFilter({
      [name] : value
    })
  }
  render() {
    return (
      <form className="Filters-container">
        <input
          type="text"
          placeholder="Search..."
          value={this.props.searchFilter}
          name="searchFilter"
          onChange={this.handleChange}
        />
        <p>
          <input
            type="checkbox"
            name="stockOnlyFitler"
            checked={this.props.stockOnlyFitler}
            onChange={this.handleChange}
          />
          Only Show products in stock
        </p>
      </form>
    );
  }
}

export default Filters;
