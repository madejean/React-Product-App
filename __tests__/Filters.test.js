import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Filters from '../src/Products/Filters';

test('Filters component should render as expected', () => {
  const component = shallow(<Filters />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
