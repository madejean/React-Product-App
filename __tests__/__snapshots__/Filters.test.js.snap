exports[`test Filters component should render as expected 1`] = `
<form
  className="Filters-container">
  <input
    name="searchFilter"
    onChange={[Function]}
    placeholder="Search..."
    type="text" />
  <p>
    <input
      name="stockOnlyFitler"
      onChange={[Function]}
      type="checkbox" />
    Only Show products in stock
  </p>
</form>
`;
